lib/awful/completion.lua

```diff
@@ -154,10 +161,25 @@ function completion.shell(command, cur_pos, ncomp, shell)
         while true do
```

Comment:
> However, we could create a small directory tree with some executable files and use a custom `PATH` while executing the consistency / sorting tests.

Yes, or patch the executable-check in case we use a custom function.

> Should I switch shells? ;-)

Only after this PR is merged.. ;)